napari/components/viewer_model.py

```diff
@@ -272,25 +270,30 @@ def _update_active_layer(self, event):
         # iteration goes backwards to find top most selected layer if any
         # if multiple layers are selected sets the active layer to None
 
+        # TODO: remove this in favor of directly setting selection.current.
         active_layer = None
         for layer in self.layers:
-            if active_layer is None and layer.selected:
+            if active_layer is None and layer in self.layers.selection:
                 active_layer = layer
-            elif active_layer is not None and layer.selected:
+            elif active_layer is not None and layer in self.layers.selection:
                 active_layer = None
                 break
 
         if active_layer is None:
             self.help = ''
             self.cursor.style = 'standard'
             self.camera.interactive = True
-            self.active_layer = None
+            self.layers.selection.current = None
         else:
             self.help = active_layer.help
             self.cursor.style = active_layer.cursor
             self.cursor.size = active_layer.cursor_size
             self.camera.interactive = active_layer.interactive
-            self.active_layer = active_layer
+            self.layers.selection.current = active_layer
+
+    @property
+    def active_layer(self):
```

Comment:
Can then deprecate, also remove/ deprecate the `active_layer` event emission too